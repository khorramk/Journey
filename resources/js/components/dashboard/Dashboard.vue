<template>
    <div class="dashboard-container bg-wood">
        <Nav />
        <div class="user-info-container dashboard relative" v-if="this.$store.state.Modal.open === false">
            <!-- <Avatar/> -->
            <div class="posts container border border-black border-2">
                <Posts :posts="this.$store.state.Posts.allPosts"/>
            </div>
            <AddButton/>
        </div>
        <Modal v-if="this.$store.state.Modal.open === true &&  this.$store.state.closeButton.close === false"/>
    </div>
</template>

<script>
import Posts from '../Posts/Posts';
import Avatar from '../dashboard/Avatar';
import Nav from '../Nav';
import AddButton from '../Visited/AddButton';
import Modal from '../Modal';
    export default {
        methods: {
            logout() {
                axios.post('/logout').then(()=> window.location.href = '/');
            }
        },
        components: {
            Posts,
            Avatar,
            Nav,
            AddButton,
            Modal
        },
        mounted () {
             axios.get('/api/posts').then((resp) => {
                console.log(resp.data);
                this.$store.dispatch('getUserPosts', resp.data.loadUsersPosts);
                this.$store.dispatch('getAllPosts', resp.data.allPosts);
                });
        },

    }
</script>
<style lang="scss" scoped>
    @mixin dashboard-size{
        height: calc(100vh - 80px);
        width: 100%;
    }
    @mixin dashboard-vis{
        background: rgb(66, 51, 51);
        overflow-y:initial;
    }
    .dashboard{
        @include dashboard-size();
        @include dashboard-vis();
    }
</style>

