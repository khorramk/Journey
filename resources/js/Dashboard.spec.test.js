import { mount } from '@vue/test-utils';
import Vue from 'vue';
import axios from 'axios';
import VueRouter from 'vue-router';
import Dashboard from './components/dashboard/Dashboard';

const vm = new Vue(Dashboard);

it('renders the elements', () => {

    vm.$nextTick().then(()=> {

     expect(vm).toMatchSnapshot();
    }).catch().finally()
  })
