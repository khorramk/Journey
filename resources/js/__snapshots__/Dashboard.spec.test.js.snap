// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the elements 1`] = `
Vue {
  "$attrs": Object {},
  "$children": Array [],
  "$createElement": [Function],
  "$listeners": Object {},
  "$options": Object {
    "_base": [Function],
    "components": Object {
      "AddButton": Object {
        "methods": Object {
          "showFullModal": [Function],
        },
        "render": [Function],
        "staticRenderFns": Array [],
      },
      "Avatar": Object {
        "data": [Function],
        "methods": Object {
          "insertAvatar": [Function],
          "onSelectFile": [Function],
          "openSelection": [Function],
          "removeSelection": [Function],
        },
        "render": [Function],
        "staticRenderFns": Array [],
      },
      "CommentModal": Object {
        "components": Object {
          "Comments": Object {
            "data": [Function],
            "mounted": [Function],
            "render": [Function],
            "staticRenderFns": Array [
              [Function],
            ],
          },
        },
        "data": [Function],
        "methods": Object {
          "submit": [Function],
        },
        "render": [Function],
        "staticRenderFns": Array [],
      },
      "Modal": Object {
        "components": Object {
          "CloseButton": Object {
            "data": [Function],
            "methods": Object {
              "clickedToClose": [Function],
            },
            "render": [Function],
            "staticRenderFns": Array [],
          },
        },
        "data": [Function],
        "methods": Object {
          "submit": [Function],
        },
        "render": [Function],
        "staticRenderFns": Array [],
      },
      "Nav": Object {
        "data": [Function],
        "methods": Object {
          "logout": [Function],
          "popLogOutBtn": [Function],
        },
        "render": [Function],
        "staticRenderFns": Array [],
      },
      "Posts": Object {
        "components": Object {
          "Avatar": Object {
            "data": [Function],
            "methods": Object {
              "insertAvatar": [Function],
              "onSelectFile": [Function],
              "openSelection": [Function],
              "removeSelection": [Function],
            },
            "render": [Function],
            "staticRenderFns": Array [],
          },
        },
        "data": [Function],
        "methods": Object {
          "flashModal": [Function],
          "submit": [Function],
        },
        "props": Object {
          "likes": Object {
            "default": [Function],
            "type": [Function],
          },
          "post": Object {
            "default": [Function],
            "type": [Function],
          },
        },
        "render": [Function],
        "staticRenderFns": Array [
          [Function],
        ],
      },
    },
    "data": [Function],
    "directives": Object {},
    "filters": Object {},
    "methods": Object {
      "logout": [Function],
    },
    "mounted": Array [
      [Function],
    ],
    "render": [Function],
    "staticRenderFns": Array [],
  },
  "$parent": undefined,
  "$refs": Object {},
  "$root": [Circular],
  "$scopedSlots": Object {},
  "$slots": Object {},
  "$vnode": undefined,
  "_c": [Function],
  "_data": Object {
    "likes": 0,
  },
  "_directInactive": false,
  "_events": Object {},
  "_hasHookEvent": false,
  "_inactive": null,
  "_isBeingDestroyed": false,
  "_isDestroyed": false,
  "_isMounted": false,
  "_isVue": true,
  "_renderProxy": Vue {
    "$attrs": Object {},
    "$children": Array [],
    "$createElement": [Function],
    "$listeners": Object {},
    "$options": Object {
      "_base": [Function],
      "components": Object {
        "AddButton": Object {
          "methods": Object {
            "showFullModal": [Function],
          },
          "render": [Function],
          "staticRenderFns": Array [],
        },
        "Avatar": Object {
          "data": [Function],
          "methods": Object {
            "insertAvatar": [Function],
            "onSelectFile": [Function],
            "openSelection": [Function],
            "removeSelection": [Function],
          },
          "render": [Function],
          "staticRenderFns": Array [],
        },
        "CommentModal": Object {
          "components": Object {
            "Comments": Object {
              "data": [Function],
              "mounted": [Function],
              "render": [Function],
              "staticRenderFns": Array [
                [Function],
              ],
            },
          },
          "data": [Function],
          "methods": Object {
            "submit": [Function],
          },
          "render": [Function],
          "staticRenderFns": Array [],
        },
        "Modal": Object {
          "components": Object {
            "CloseButton": Object {
              "data": [Function],
              "methods": Object {
                "clickedToClose": [Function],
              },
              "render": [Function],
              "staticRenderFns": Array [],
            },
          },
          "data": [Function],
          "methods": Object {
            "submit": [Function],
          },
          "render": [Function],
          "staticRenderFns": Array [],
        },
        "Nav": Object {
          "data": [Function],
          "methods": Object {
            "logout": [Function],
            "popLogOutBtn": [Function],
          },
          "render": [Function],
          "staticRenderFns": Array [],
        },
        "Posts": Object {
          "components": Object {
            "Avatar": Object {
              "data": [Function],
              "methods": Object {
                "insertAvatar": [Function],
                "onSelectFile": [Function],
                "openSelection": [Function],
                "removeSelection": [Function],
              },
              "render": [Function],
              "staticRenderFns": Array [],
            },
          },
          "data": [Function],
          "methods": Object {
            "flashModal": [Function],
            "submit": [Function],
          },
          "props": Object {
            "likes": Object {
              "default": [Function],
              "type": [Function],
            },
            "post": Object {
              "default": [Function],
              "type": [Function],
            },
          },
          "render": [Function],
          "staticRenderFns": Array [
            [Function],
          ],
        },
      },
      "data": [Function],
      "directives": Object {},
      "filters": Object {},
      "methods": Object {
        "logout": [Function],
      },
      "mounted": Array [
        [Function],
      ],
      "render": [Function],
      "staticRenderFns": Array [],
    },
    "$parent": undefined,
    "$refs": Object {},
    "$root": [Circular],
    "$scopedSlots": Object {},
    "$slots": Object {},
    "$vnode": undefined,
    "_c": [Function],
    "_data": Object {
      "likes": 0,
    },
    "_directInactive": false,
    "_events": Object {},
    "_hasHookEvent": false,
    "_inactive": null,
    "_isBeingDestroyed": false,
    "_isDestroyed": false,
    "_isMounted": false,
    "_isVue": true,
    "_renderProxy": [Circular],
    "_self": [Circular],
    "_staticTrees": null,
    "_uid": 0,
    "_vnode": null,
    "_watcher": null,
    "_watchers": Array [],
    "likes": 0,
    "logout": [Function],
  },
  "_self": [Circular],
  "_staticTrees": null,
  "_uid": 0,
  "_vnode": null,
  "_watcher": null,
  "_watchers": Array [],
  "likes": 0,
  "logout": [Function],
}
`;
